src/core/prompts/sections/custom-instructions.ts

```diff
@@ -16,7 +19,81 @@ async function safeReadFile(filePath: string): Promise<string> {
 	}
 }
 
+/**
+ * Check if a directory exists
+ */
+async function directoryExists(dirPath: string): Promise<boolean> {
+	try {
+		const stats = await fs.stat(dirPath)
+		return stats.isDirectory()
+	} catch (err) {
+		return false
+	}
+}
+
+/**
+ * Read all text files from a directory in alphabetical order
+ */
+async function readTextFilesFromDirectory(dirPath: string): Promise<Array<{ filename: string; content: string }>> {
+	try {
+		const files = await fs
+			.readdir(dirPath, { withFileTypes: true, recursive: true })
+			.then((files) => files.filter((file) => file.isFile()))
+			.then((files) => files.map((file) => path.resolve(dirPath, file.name)))
+
+		const fileContents = await Promise.all(
+			files.map(async (file) => {
+				const filePath = path.join(dirPath, file)
+				try {
+					// Check if it's a file (not a directory)
+					const stats = await fs.stat(filePath)
+					if (stats.isFile()) {
+						const content = await safeReadFile(filePath)
+						return { filename: file, content }
+					}
+					return null
+				} catch (err) {
+					return null
+				}
+			}),
+		)
+
+		// Filter out null values (directories or failed reads)
+		return fileContents.filter((item): item is { filename: string; content: string } => item !== null)
+	} catch (err) {
+		return []
+	}
+}
+
+/**
+ * Format content from multiple files with filenames as headers
+ */
+function formatDirectoryContent(dirPath: string, files: Array<{ filename: string; content: string }>): string {
+	if (files.length === 0) return ""
+
+	const formattedContent = files
+		.map((file) => {
+			return `# Filename: ${file.filename}\n${file.content}`
+		})
+		.join("\n\n")
+
+	return `\n# Rules from ${path.relative(process.cwd(), dirPath)}:\n${formattedContent}\n`
+}
+
+/**
+ * Load rule files from the specified directory
+ */
 export async function loadRuleFiles(cwd: string): Promise<string> {
+	// Check for .roo/rules/ directory
+	const rooRulesDir = path.join(cwd, ".roo", "rules")
+	if (await directoryExists(rooRulesDir)) {
+		const files = await readTextFilesFromDirectory(rooRulesDir)
+		if (files.length > 0) {
+			return formatDirectoryContent(rooRulesDir, files)
```

Comment:
One thing I'm not sure about - maybe we should include both the files in .roo/rules/ and .roorules if present?

Reply:
Or maybe that's overcomplicating things, not sure.

Reply:
I also thought about it, but felt it was complicated, and wanted to load only one side.

Since Cline uses either the `.clinerules/` directory or `.clinrules` file, it would be clear if Roo-Code follows the same specification. (In Cline's case, there is also a technical constraint that directories and files with the same name cannot coexist...)

Reply:
Sounds good!